File: Backend/Entities/MissionOrg.cs
using System;
using System.Collections.Generic;
using LinqToDB;
using LinqToDB.Mapping;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace Backend.Entities
{
    public class MissionOrg : BaseEntity
    {
        public string Name { get; set; }
        public Guid RepId { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; }
        public string Address { get; set; }
        public string AddressLocal { get; set; }
        public bool OfficeInThailand { get; set; }
        public DateTime? ApprovedDate { get; set; }

        [Column(DataType = DataType.VarChar)]
        public MissionOrgStatus? Status { get; set; }
    }

    public class MissionOrgWithNames : MissionOrg
    {
        public string ContactName { get; set; }
    }

    public class MissionOrgWithYearSummaries : MissionOrg
    {
        public IList<MissionOrgYearSummary> YearSummaries { get; set; }
    }

    [JsonConverter(typeof(StringEnumConverter))]
    public enum MissionOrgStatus
    {
        [MapValue(nameof(Associate))]
        Associate,

        [MapValue(nameof(OwnerAssociate))]
        OwnerAssociate,

        [MapValue("FounderAssociate")]
        [MapValue(nameof(FoundingAssociate))]
        FoundingAssociate,

        [MapValue(nameof(Founder))]
        Founder
    }

    public class MissionOrgYearSummary : BaseEntity
    {
        public Guid MissionOrgId { get; set; }
        public int Year { get; set; }
        public int StudentCount { get; set; }
        public int TeacherCount { get; set; }

        [Column(DataType = DataType.VarChar)]
        public MissionOrgStatus? Status { get; set; }

        [Column(DataType = DataType.VarChar)]
        public MissionOrgLevel? Level { get; set; }
    }

    [JsonConverter(typeof(StringEnumConverter))]
    public enum MissionOrgLevel
    {
        [MapValue(nameof(Bronze))]
        Bronze,

        [MapValue(nameof(Silver))]
        Silver,

        [MapValue(nameof(Gold))]
        Gold,

        [MapValue(nameof(Platinum))]
        Platinum
    }
}

File: Frontend/src/app/mission-org/mission-org.ts
import { BaseEntity } from '../classes/base-entity';
import { MissionOrgYearSummary } from './mission-org-year-summary';

export class MissionOrg extends BaseEntity {
  public name: string;
  public repId: string;
  public phone: string;
  public email: string;
  public address: string;
  public addressLocal: string;
  public officeInThailand: boolean;
  public approvedDate: Date;
  public status?: MissionOrgStatus;
}

export class MissionOrgWithNames extends MissionOrg {
  public contactName: string;
}

export class MissionOrgWithYearSummaries extends MissionOrg {
  public yearSummaries: MissionOrgYearSummary[] = [];
}


export enum MissionOrgStatus {
  Associate = 'Associate',
  OwnerAssociate = 'OwnerAssociate',
  FoundingAssociate = 'FoundingAssociate',
  Founder = 'Founder'
}
